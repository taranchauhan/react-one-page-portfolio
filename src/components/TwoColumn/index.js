import React from 'react';
import styled from 'styled-components';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

import Projects from 'containers/Projects';
import About from 'containers/About';
import Contact from 'containers/Contact';

const Row = styled.div`
  display: flex;
  height: 100vh;
`;

const Column = styled.div`
  flex: 50%;
  padding: 10px;
  border: 1px solid;
`;

const Ul = styled.ul`
  list-style: none;
`;

const Li = styled.li``;

const StyledLink = styled(Link)``;

const TwoColumn = () => (
  <Router>
    <Row>
      <Column>
        <h1>Portfolio</h1>
        <h2>Subtitle</h2>
        <Ul>
          <Li>
            <StyledLink to="/">Projects</StyledLink>
          </Li>
          <Li>
            <StyledLink to="/about">About</StyledLink>
          </Li>
          <Li>
            <StyledLink to="/contact">Contact</StyledLink>
          </Li>
        </Ul>
      </Column>
      <Column>
        <Route exact path="/" component={Projects} />
        <Route exact path="/about" component={About} />
        <Route exact path="/contact" component={Contact} />
      </Column>
    </Row>
  </Router>
);

export default TwoColumn;
