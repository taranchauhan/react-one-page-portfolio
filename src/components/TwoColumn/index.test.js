import React from 'components/TwoColumn/react';
import { shallow } from 'components/TwoColumn/enzyme';
import TwoColumn from './index';

describe('<TwoColumn />', () => {
  it('should render the TwoColumn', () => {
    const wrapper = shallow(<TwoColumn />);
    expect(wrapper).toMatchSnapshot();
  });
});
