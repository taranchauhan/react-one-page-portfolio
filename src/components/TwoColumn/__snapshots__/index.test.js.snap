// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwoColumn /> should render the TwoColumn 1`] = `
<BrowserRouter>
  <styled.div>
    <styled.div
      flex="30%"
    >
      <LeftColumnContent />
    </styled.div>
    <styled.div
      flex="70%"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
        path="/contact"
      />
    </styled.div>
  </styled.div>
</BrowserRouter>
`;
